File: backend/passatempo/src/main/java/com/luisn/passatempo/domain/Titulo.java
package com.luisn.passatempo.domain;

import jakarta.persistence.*;
import lombok.Data;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
public class Titulo {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    private String name;

    private int ano;

    private String sinopse;

    private String categoria;

    @ManyToMany
    @JoinTable(name = "ator_titulo",
            joinColumns = @JoinColumn(name = "id_titulo"),
            inverseJoinColumns = @JoinColumn(name = "id_ator"))
    private List<Ator> listaAtores = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "id_diretor")
    private Diretor diretor;

    @ManyToOne
    @JoinColumn(name = "id_classe")
    private Classe classe;

    @OneToMany(mappedBy = "titulo")
    private List<Item> listaItens;
}

File: frontend/passatempo/src/app/models/titulo.ts
import { Ator } from "./ator";
import { Classe } from "./classe";
import { Diretor } from "./diretor";

export class Titulo {
   constructor (
      public id?: string,
      public name?: string,
      public ano?: number,
      public sinopse?: string,
      public categoria?: string,
      public listaAtores?: Ator[],
      public diretor?: Diretor,
      public classe?: Classe
   ) {}
}
["Suspense","Terror","Comédia","Ação",
"Aventura","Drama", "Comédia Romântica", "Ficção Científica"]

export const categorias = [
   {id: '1', name: 'Suspense'},
   {id: '2', name: 'Terror'},
   {id: '3', name: 'Comédia'},
   {id: '4', name: 'Ação'},
   {id: '5', name: 'Aventura'},
   {id: '6', name: 'Drama'},
   {id: '7', name: 'Comédia Romântica'},
   {id: '8', name: 'Ficção Científica'}
 ]
 